XcodeServerSDK/XcodeServerConfig.swift

```diff
@@ -33,16 +42,51 @@ public class XcodeServerConfig : JSONSerializable {
         return dict
     }
     
-    public init(var host: String, user: String?, password: String?) {
+    ///
+    /// Initializes a server configuration with the provided host.
+    /// - parameter host: `Xcode` server host.
+    /// - paramater user: Username that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - parameter password: Password that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - returns: A fully initialized `XcodeServerConfig` instance.
+    ///
+    /// - throws:
+    ///     - `InvalidHostProvided`: When the host provided doesn't produce a valid `URL`
+    ///     - `InvalidSchemeProvided`: When the provided scheme is not `HTTPS`
+    ///
+    public required init(var host: String, user: String?, password: String?) throws {
+        guard let url = NSURL(string: host) else {
+            /*******************************************************************
```

Comment:
(I think you mean `init did not throw` :blush:)

But yeah I like this. I guess the paradigm is... I want to test calling one function/block and I want to make sure it throws one specific type of an error. In that case, I want to print that it all went well, otherwise fail the test and print what actual error got returned.

Can you think of an easy function to do this for us to avoid the repetition? We already have `TestUtils.swift` if you end up coming up with something.